drivers/bridge/bridge.go

```diff
@@ -46,6 +42,8 @@ type Configuration struct {
 	EnableICC             bool
 	EnableIPForwarding    bool
 	AllowNonDefaultBridge bool
+	// Driver components
+	PortAllocator *portallocator.PortAllocator
```

Comment:
Also, the proposed implementation leaves it open to later have separate port allocators with no change to the existing infrastructure, we pass pa transparenlty to the netwrok driver as part of the Config. If we ever need to change this, it is going to be a minimally intrusive change.

What you are proposing instead, that is making it a singleton, set us upfront for invasive infrastructure changes if we ever need new separate instances of pa.

Again, the requirement here is to do proper separation of concerns about existing portallocator. Current port allocator is providing host wide information book-keeping, which is needed by docker core as well by libnetwork. This does not mean port allocator needs to be singleton, even though this would serve the purpose, because it would limit future use of it which we are not not aware of.

I don't think at this point we can make the call about port allocator be a singleton package.